client/wc-api/items/mutations.js

```diff
@@ -1,15 +1,47 @@
 /** @format */
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { dispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { getResourceName } from '../utils';
 
-const updateItem = operations => ( type, id, itemData ) => {
-	const resourceName = getResourceName( `items-query-${ type }-item`, id );
-	operations.update( [ resourceName ], { [ resourceName ]: { id, ...itemData } } );
+const updateProductStock = operations => async ( product, newStock ) => {
+	const { addNotice } = dispatch( 'wc-admin' );
+	const oldStockQuantity = product.stock_quantity;
+	const resourceName = getResourceName( 'items-query-products-item', product.id );
+
+	// Optimistically update product stock
+	operations.updateLocally( [ resourceName ], {
+		[ resourceName ]: { ...product, stock_quantity: newStock },
+	} );
+
+	const result = await operations.update( [ resourceName ], {
+		[ resourceName ]: { id: product.id, stock_quantity: newStock },
```

Comment:
`type` and ` parent_id` need to be passed in here as well. There is a switch statement in `update` that requires it.

https://github.com/woocommerce/woocommerce-admin/blob/b19643ed72715a045d9114123789d222933a0497/client/wc-api/items/operations.js#L98-L106

Reply:
Oh, right I forgot that when refactoring :man_facepalming: , thanks for noticing!

It's fixed in 855587a72f9baca96d4f2c4210495b3d3ebb7884.